controllers/openstackbaremetalset_controller.go

```diff
@@ -1017,15 +1017,21 @@ func (r *OpenStackBaremetalSetReconciler) baremetalHostProvision(
 		)
 
 		//
-		// Update the BMH spec once when ConsumerRef is nil to only perform one time provision.
+		// Ensure the image url is up to date unless already provisioned
 		//
-		if foundBaremetalHost.Spec.ConsumerRef == nil {
-			foundBaremetalHost.Spec.Online = true
-			foundBaremetalHost.Spec.ConsumerRef = &corev1.ObjectReference{Name: instance.Name, Kind: instance.Kind, Namespace: instance.Namespace}
+		if string(foundBaremetalHost.Status.Provisioning.State) != "provisioned" {
 			foundBaremetalHost.Spec.Image = &metal3v1alpha1.Image{
 				URL:      localImageURL,
 				Checksum: fmt.Sprintf("%s.md5sum", localImageURL),
 			}
+		}
+
+		//
+		// Update the BMH spec once when ConsumerRef is nil to only perform one time provision.
+		//
+		if foundBaremetalHost.Spec.ConsumerRef == nil {
+			foundBaremetalHost.Spec.Online = true
+			foundBaremetalHost.Spec.ConsumerRef = &corev1.ObjectReference{Name: instance.Name, Kind: instance.Kind, Namespace: instance.Namespace}
 			foundBaremetalHost.Spec.UserData = &corev1.SecretReference{
```

Comment:
Maybe the secrets references should also be updated until the host is provisioned?